### PR TITLE
MULTIARCH-5172: Add Power VS MAPI and CAPI test fixtures

### PR DESCRIPTION
**Pull request by Karthik-K-N**
Added Power VS CAPI and MAPI test fixtures.

### CODE REVIEW

**Comment by Karthik-K-N** [comment]
@JoelSpeed @damdo Please take a look when you get chance. Thanks

File: testutils/resourcebuilder/cluster-api/infrastructure/v1beta2/powervscluster.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+func (p PowerVSClusterBuilder) WithResourceGroup(resourceGroup *capibmv1.IBMPowerVSResourceReference) PowerVSClusterBuilder {
+	p.resourceGroup = resourceGroup
+	return p
+}
+
+// WithLoadBalancer sets the loadBalancers for the IBMPowerVSCluster builder.
+func (p PowerVSClusterBuilder) WithLoadBalancer(loadBalancers []capibmv1.VPCLoadBalancerSpec) PowerVSClusterBuilder {
+	p.loadBalancers = loadBalancers
+	return p
+}
```
Comment (JoelSpeed) [inline testutils/resourcebuilder/cluster-api/infrastructure/v1beta2/powervscluster.go:165]: Can you sort the fields alphabetically within each group, this should then be consistent with other providers

File: testutils/resourcebuilder/cluster-api/infrastructure/v1beta2/powervsmachine.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+func (p PowerVSMachineBuilder) WithNetwork(network capibmv1.IBMPowerVSResourceReference) PowerVSMachineBuilder {
+	p.network = network
+	return p
+}
+
+// WithProviderID sets the providerID for the PowerVSMachine builder.
+func (p PowerVSMachineBuilder) WithProviderID(providerID *string) PowerVSMachineBuilder {
+	p.providerID = providerID
+	return p
+}
```
Comment (JoelSpeed) [inline testutils/resourcebuilder/cluster-api/infrastructure/v1beta2/powervsmachine.go:185]: Please sort alphabetically

File: testutils/resourcebuilder/cluster-api/infrastructure/v1beta2/powervsmachine.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+func (p PowerVSMachineBuilder) WithFailureReason(failureReason *errors.MachineStatusError) PowerVSMachineBuilder {
+	p.failureReason = failureReason
+	return p
+}
+
+// WithConditions sets the conditions for the PowerVSMachine builder.
+func (p PowerVSMachineBuilder) WithConditions(conditions clusterv1.Conditions) PowerVSMachineBuilder {
+	p.conditions = conditions
+	return p
+}
```
Comment (JoelSpeed) [inline testutils/resourcebuilder/cluster-api/infrastructure/v1beta2/powervsmachine.go:229]: Please sort alphabetically

File: testutils/resourcebuilder/cluster-api/infrastructure/v1beta2/powervsmachinetemplate.go [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+func (p PowerVSMachineTemplateBuilder) WithNetwork(network capibmv1.IBMPowerVSResourceReference) PowerVSMachineTemplateBuilder {
+	p.network = network
+	return p
+}
+
+// WithProviderID sets the providerID for the PowerVSMachineTemplate builder.
+func (p PowerVSMachineTemplateBuilder) WithProviderID(providerID *string) PowerVSMachineTemplateBuilder {
+	p.providerID = providerID
+	return p
+}
```
Comment (JoelSpeed) [inline testutils/resourcebuilder/cluster-api/infrastructure/v1beta2/powervsmachinetemplate.go:200]: Please sort alphabetically

File: testutils/resourcebuilder/machine/v1/powervs_provider_spec.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+)
+
+var (
+	defaultUserDataSecretName          = "powervs-user-data-12345678"
+	defaultCredentialsSecretName       = "powervs-cloud-credentials"
+	defaultKeyPairName                 = "powervs-key-12345678"
+	defaultSystemType                  = "s922"
+	defaultProcessorType               = machinev1.PowerVSProcessorTypeShared
+	defaultProcessors                  = intstr.FromString("1")
+	defaultMemory                int32 = 32
```
Comment (JoelSpeed) [inline testutils/resourcebuilder/machine/v1/powervs_provider_spec.go:37]: Misaligned

File: testutils/resourcebuilder/machine/v1/powervs_provider_spec.go [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+		// As we are building the input to json.Marshal, this should never happen.
+		panic(err)
+	}
+
+	return &runtime.RawExtension{
+		Raw: raw,
+	}
+}
+
+// WithUserDataSecret sets the userDataSecret for the PowerVS machine config builder.
```
Comment (JoelSpeed) [inline testutils/resourcebuilder/machine/v1/powervs_provider_spec.go:114]: Please sort all the builder functions alphabetically

**Comment by Karthik-K-N** [comment]
Sorted all the methods and fields alphabetically. Please take a look.

**Comment by damdo** [comment]
Thanks @Karthik-K-N for working on this!

**Pull request merged**